app/src/main/java/com/duckduckgo/app/browser/httperrors/HttpErrorPixels.kt

```diff
@@ -18,32 +18,86 @@ package com.duckduckgo.app.browser.httperrors
+import com.duckduckgo.app.pixels.remoteconfig.AndroidBrowserConfigFeature
 import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.browser.api.WebViewVersionProvider
 import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.vpn.VpnFeaturesRegistry
+import com.duckduckgo.networkprotection.api.NetworkProtectionState
+import com.duckduckgo.networkprotection.api.NetworkProtectionState.ConnectionState
+import com.duckduckgo.subscriptions.api.Product.NetP
+import com.duckduckgo.subscriptions.api.SubscriptionStatus
+import com.duckduckgo.subscriptions.api.Subscriptions
 import com.squareup.anvil.annotations.ContributesBinding
 import java.time.Instant
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
 
 interface HttpErrorPixels {
     fun updateCountPixel(httpErrorPixelName: HttpErrorPixelName)
+    suspend fun update5xxCountPixel(httpErrorPixelName: HttpErrorPixelName, statusCode: Int)
     fun fireCountPixel(httpErrorPixelName: HttpErrorPixelName)
+    fun fire5xxCountPixels()
 }
 
 @ContributesBinding(AppScope::class)
 class RealHttpErrorPixels @Inject constructor(
     private val pixel: Pixel,
     private val context: Context,
+    private val webViewVersionProvider: WebViewVersionProvider,
+    private val networkProtectionState: NetworkProtectionState,
+    private val subscriptions: Subscriptions,
+    private val vpnFeaturesRegistry: VpnFeaturesRegistry,
+    private val androidBrowserConfig: AndroidBrowserConfigFeature,
 ) : HttpErrorPixels {
 
     private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
+    private val pixel5xxKeys: MutableSet<String> by lazy {
+        preferences.getStringSet(PIXEL_5XX_KEYS_SET, mutableSetOf()) ?: mutableSetOf()
+    }
 
     override fun updateCountPixel(httpErrorPixelName: HttpErrorPixelName) {
         val count = preferences.getInt(httpErrorPixelName.appendCountSuffix(), 0)
         preferences.edit { putInt(httpErrorPixelName.appendCountSuffix(), count + 1) }
     }
 
+    override suspend fun update5xxCountPixel(
+        httpErrorPixelName: HttpErrorPixelName,
+        statusCode: Int,
+    ) {
+        // Kill switch
+        if (!androidBrowserConfig.self().isEnabled() || !androidBrowserConfig.httpError5xxPixel().isEnabled()) {
+            return
+        }
+
+        combine(
+            subscriptions.getEntitlementStatus().map { entitledProducts -> entitledProducts.contains(NetP) },
+            networkProtectionState.getConnectionStateFlow(),
+            flowOf(webViewVersionProvider.getFullVersion()),
+        ) { netpEntitlementStatus, connectionState, webViewFullVersion ->
+            val subscriptionStatus = subscriptions.getSubscriptionStatus()
+            val pProVpnConnected = isVpnConnected(netpEntitlementStatus, connectionState, subscriptionStatus)
```

Comment:
@anikiki we could just use: `networkProtectionState.isRunning` here and return a boolean.